ferrocene/doc/internal-procedures/src/testing-other-targets.rst

```diff
@@ -0,0 +1,193 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Ferrocene Developers
+
+Testing Other Targets 
+=====================
+
+It's often possible to test targets other than your host tuple. Windows and macOS hosts can target
+Linux using `Lima <https://lima-vm.io/>`_ (Mac/Linux) or
+`WSL2 <https://learn.microsoft.com/en-us/windows/wsl/install>`_ (Windows). Lima can be used to test
+other architectures such as :ref:`x86_64-unknown-linux-gnu`, :target:`aarch64-unknown-linux-gnu`, or
+:target:`riscv64gc-unknown-linux-gnu`, while WSL2 can do similar with `QEMU <https://www.qemu.org/>`_.
+
+Additionally, Ferrocene supports testing a number of targets which are not supported by upstream.
+When testing locally, special tools or configuration may be required.
+
+In general, any "bare-metal" target listed in :doc:`user-manual:targets/index` requires special
+setup inside a Linux based environment.
+
+Host Setup
+----------
+
+Unless otherwise noted, all bare-metal targets are tested in QEMU on a Linux host.
+On macOS, a tool like Lima or Docker must be used. On Windows, WSL2 must be used.
+
+Please refer to the relevant per-host setup instructions below.
+
+:ref:`x86_64-unknown-linux-gnu`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You need to have all the normal prerequisites for testing Rust installed, as well as a few extras:
+
+.. code-block:: bash
+
+   sudo apt install awscli ninja-build bzip2 cmake gcc g++ qemu-user-static binfmt-support
```

Comment:
AWS CLI should not be installed through apt, at least on Ubuntu.

We require version 2.x of the CLI (due to our use of AWS SSO), but before Ubuntu 24.04 Ubuntu only shipped version 1.x. Then, in 24.04 they removed awscli from the archives (replacing it with a snap package), and then 2.x seems to be available in 24.10 now.

We already have it listed as a prerequisite in the onboarding instructions, so that should be enough. I'd love the whole list of packages (minus awscli) to be moved to the onboarding instructions.

Reply:
Some packages are testing specific (eg binfmt) so I'd keep those isolated here.

I glanced through onboarding and it seems like `setup-local-env.rst` is a more appropriate place (it already lists some dependencies). I'll update that page with per-OS dep lists.